src/subtitleTrack.ts

```diff
@@ -48,13 +46,40 @@ export interface SubtitleTrack {
 }
 
 /**
- * Helper type that represents an entry in `SourceConfig.subtitleTracks` list.
+ * A subtitle track that can be added to `SourceConfig.subtitleTracks`.
  *
- * Since `SubtitleTrack` has all of its properties as optionals for total compatibility with
- * values that may be sent from native code, this type serves as a reinforcer of what properties
- * should be required during the registration of an external subtitle track from JS.
  */
-export type SideLoadedSubtitleTrack = MakeRequired<
-  SubtitleTrack,
-  'url' | 'label' | 'language' | 'format'
->;
+export interface SideLoadedSubtitleTrack {
```

Comment:
I'm not a very big fan of documentation and definition duplication that we have here now between SubtitleTrack and SideLoadedSubtitleTrack. 
Would be good if we could achieve some other way of making some fields not optional. I'm no TS expert but I just tried to extend the SubtitleTack interface and it looks like it achieves what we want here:

![image](https://github.com/bitmovin/bitmovin-player-react-native/assets/11734608/c0ff7d49-a1a8-4b9f-a82b-8a2430c3fe6b)

On the calling side I now get an error if I don't specify a non optional property: 
![image](https://github.com/bitmovin/bitmovin-player-react-native/assets/11734608/ed268dab-f374-4006-ae8c-9287cd6e6feb)

What do you think about this approach?

Reply:
Thanks, this is a great idea!
Implemented in c73c55d

Note: looks like the documentation gets picked up by the doc gen tool and the IDE as well, so we don't have to duplicate it.